diff -rupN DPO_RT2870_LinuxSTA_V2.3.0.0.old/os/linux/config.mk DPO_RT2870_LinuxSTA_V2.3.0.0/os/linux/config.mk
--- DPO_RT2870_LinuxSTA_V2.3.0.0.old/os/linux/config.mk	2009-12-09 01:15:38.000000000 -0500
+++ DPO_RT2870_LinuxSTA_V2.3.0.0/os/linux/config.mk	2009-12-09 01:17:53.000000000 -0500
@@ -8,10 +8,10 @@ HAS_QA_SUPPORT=n
 HAS_XLINK=n
 
 # Support Wpa_Supplicant
-HAS_WPA_SUPPLICANT=n
+HAS_WPA_SUPPLICANT=y
 
 # Support Native WpaSupplicant for Network Maganger
-HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=n
+HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=y
 
 #Support Net interface block while Tx-Sw queue full
 HAS_BLOCK_NET_IF=n
